nemo/utils/model_utils.py

```diff
@@ -0,0 +1,237 @@
+    multiple dataloaders for the validation set.
+
+    It does so by first resolving the path to one/more data files via `resolve_filepath_from_cfg()`.
+    If this resolution fails, it assumes the data loader is prepared to manually support / not support
+    multiple data loaders and simply calls the appropriate setup method.
+
+    If resolution succeeds:
+        Checks if provided path is to a single file or a list of files.
+        If a single file is provided, simply tags that file as such and loads it via the setup method.
+        If multiple files are provided:
+            Inject a new manifest path at index "i" into the resolved key.
+            Calls the appropriate setup method to set the data loader.
+            Collects the initialized data loader in a list and preserves it.
+            Once all data loaders are processed, assigns the list of loaded loaders to the ModelPT.
+            Finally assigns a list of unique names resolved from the file paths to the ModelPT.
+
+    Args:
+        model: ModelPT subclass, which requires >=1 Validation Dataloaders to be setup.
+    """
+    cfg = copy.deepcopy(model._cfg)
+    dataloaders = []
+
+    # process val_loss_idx
+    if hasattr(cfg.validation_ds, 'val_loss_idx'):
+        cfg = OmegaConf.to_container(cfg)
+        val_loss_idx = cfg['validation_ds'].pop('val_loss_idx')
+        cfg = OmegaConf.create(cfg)
+    else:
+        val_loss_idx = 0
+
+    # Set val_loss_idx
+    model._validation_loss_idx = val_loss_idx
+
+    ds_key = resolve_dataset_name_from_cfg(cfg.validation_ds)
+
+    if ds_key is None:
+        logging.debug(
+            "Could not resolve file path from provided config - {}. "
+            "Disabling support for multi-dataloaders.".format(cfg.validation_ds)
+        )
+
+        model.setup_validation_data(cfg.validation_ds)
+        return
+
+    ds_values = cfg.validation_ds[ds_key]
+
+    if isinstance(ds_values, (list, tuple, ListConfig)):
+
+        for ds_value in ds_values:
+            cfg.validation_ds[ds_key] = ds_value
+            model.setup_validation_data(cfg.validation_ds)
+            dataloaders.append(model._validation_dl)
+
+        model._validation_dl = dataloaders
+        model._validation_names = [parse_dataset_as_name(ds) for ds in ds_values]
+
+        # In fast-dev-run, only one data loader is used
+        if model._trainer.fast_dev_run:
+            model._validation_dl = model._validation_dl[:1]
+            model._validation_names = model._validation_names[:1]
+
+        return
+
+    else:
+        model.setup_validation_data(cfg.validation_ds)
+        model._validation_names = [parse_dataset_as_name(ds_values)]
+
+
+def resolve_test_dataloaders(model: 'ModelPT'):
+    """
+    Helper method that operates on the ModelPT class to automatically support
+    multiple dataloaders for the test set.
+
+    It does so by first resolving the path to one/more data files via `resolve_filepath_from_cfg()`.
```

Comment:
should `resolve_filepath_from_cfg` be `resolve_dataset_name_from_cfg`?

Reply:
Yes it should forgot to rename docstring